File: works/js/4-10/1300.js
let fs = require('fs');
let input = fs.readFileSync(0).toString().trim().split('\n')
// let input = fs.readFileSync('input.txt').toString().split('\n')

// let arr = new Array(1200).fill(0).map(() => new Array(2400).fill(' '))
let N = parseInt(input[0])
let K = parseInt(input[1])

let left = 1, right = N * N 
while(left <= right) {
    let mid = Math.floor((left + right) / 2) 

    let cnt = Count(mid) 

    if(cnt < K) {
        left = mid + 1
    }
    else {
        right = mid - 1
        answer = mid 
    }
}

console.log(answer) 


function Count(mid) {
    let cnt = 0 

    for(i = 1; i <= N; i++) {
        if(i * N <= mid) {
            cnt += N 
        }
        else {
            cnt += Math.floor(mid / i) 
        }
    }

    return cnt 
}

File: baekjoon/1300.cpp
#include <iostream>
using namespace std;

long long N, K;
long long answer;

long long Count(long long mid) {
    long long cnt = 0;

    for(int i = 1; i <= N; i++) {
        if (i * N <= mid) {
            cnt += N;
        }
        else {
            cnt += mid / i;
        }
    }

    return cnt;
}

int main() {
    cin >> N >> K;

    long long left = 1, right = N*N;
    while(left <= right) {
        long long mid = (left + right) / 2;

        long long cnt = Count(mid);

        if(cnt < K) {
            left = mid + 1;
        }
        else {
            right = mid - 1;
            answer = mid;
        }
    }

    cout << answer;
}